integration/regression_tests.sh

```diff
@@ -0,0 +1,200 @@
+#!/bin/bash
+
+# Basic binary compatibility test script for um2nc.
+# See INTEGRATION_README.md for details on test usage, data and options.
+#
+# -----------------------------------------------------------------
+# Prepare options
+# -----------------------------------------------------------------
+
+function usage {
+    echo "Basic binary compatibility test script for um2nc."
+    echo "Compares um2nc output against previous versions."
+    echo
+    echo "Usage: regression_tests.sh -o OUTPUT_DIR [-d DATA_CHOICE] [-v DATA_VERSION]"
+    echo
+    echo "Options"
+    echo "-o        Directory for writing netCDF output."
+    echo "-d        Choice of test reference data. Options: \"full\", \"intermediate\","
+    echo "          and \"light\". View INTEGRATION_README.md for details."
+    echo "          Default: \"intermediate\""
+    echo "-v        Version of test reference data to use. Options: \"v0\"."
+    echo "          View INTEGRATION_README.md for details."
+    echo "          Default: \"v0\""
+}
+
+TEST_DATA_PARENT_DIR=/g/data/vk83/testing/um2nc/integration-tests
+
+# Default values, overwritten by command line arguments if present:
+TEST_DATA_CHOICE_DEFAULT=intermediate
+TEST_DATA_VERSION_DEFAULT=v0
+
+while getopts ":-:d:ho:v:" opt; do
+    case ${opt} in
+        -)
+            case ${OPTARG} in
+                help)
+                    usage
+                    exit 0
+                ;;
+                *)
+                    echo "Invalid \"--${OPTARG}\" option." >&2
+                    usage
+                    exit 1
+                ;;
+            esac
+        ;;
+        d)
+            case ${OPTARG} in
+                full|intermediate|light)
+                    TEST_DATA_CHOICE=${OPTARG}
+                    ;;
+                *)
+                    echo "Invalid \"-${opt}\" option. Choose between \"full\", \"intermediate\" and \"light\"." >&2
+                    usage
+                    exit 1
+                ;;
+            esac
+        ;;
+        h)
+            usage
+            exit 0
+        ;;
+        o)
+            OUTPUT_DIR=${OPTARG}
+        ;;
+        v)
+            DATA_VERSION=${OPTARG}
+        ;;
+        :)
+            echo "Option \"-${OPTARG}\" requires an argument." >&2
+            usage
+            exit 1
+        ;;
+        \?)
+            echo "Invalid option: -${OPTARG}" >&2
+            usage
+            exit 1
+        ;;
+    esac
+done
+
+# Check options are valid
+if [ -z "${OUTPUT_DIR}" ]; then
+    echo "ERROR: output directory must be set using \"-o\"." >&2
+    usage
+    exit 1
+fi
```

Comment:
That's a good point. I like the idea of using a temporary directory – what would you think of removing the `-o` option entirely, so that a temporary directory is always used?

We could replace it with a `-k` (keep) flag to keep the output at the end of successful tests.

I feel like this would be slightly simpler, and there'd be zero risk of deleting anything that shouldn't be when running `rm -r $OUTPUT_DIR`. (We can try and guard against removing a user's directory e.g. with the `CLEAN_OUTPUT=false` setting but I'm always paranoid that things could go wrong.

Reply:
I like your idea to always use a temporary directory and give the option to keep the output files after a successful run with `-k`/`--keep`.
Let's do that! :)

Reply:
Sweet! I've made these changes in b8afff3